app/torii-providers/arcgis-oauth-bearer.js

```diff
@@ -39,6 +42,36 @@
    _currentBaseUrl: function () {
      return [window.location.protocol, '//', window.location.host].join('');
    },
+
+   buildQueryString: function(options){
+     const requiredParams = this.get('requiredUrlParams');
+     const optionalParams = this.get('optionalUrlParams');
+
+     const qs = QueryString.create({
+       provider: this,
+       requiredParams: requiredParams,
+       optionalParams: optionalParams,
+       options
+     });
+
+     return qs.toString();
+  },
+
+  buildUrl: function(options){
+    let base = this.get('baseUrl');
+    if (options.portalUrl || options.path) {
+      base = options.portalUrl || this.get('portalUrl');
+      const path = options.path || this.get('path');
+      base = `${base}${path}`;
+    }
+    delete options.portalUrl;
+    delete options.path;
+
+    const qs = this.buildQueryString(options);
+
+    return [base, qs].join('?');
+  },
```

Comment:
This is the same as the torii default implementation except that it uses `options.portal` and `options.path` to construct the base url if they are provided and it passes `options` on to `buildQueryString`.